components/org.wso2.carbon.identity.scim2.common/src/main/java/org/wso2/carbon/identity/scim2/common/impl/SCIMUserManager.java

```diff
@@ -522,7 +523,17 @@ private long getTotalUsers(String domainName) throws CharonException {
             try {
                 AbstractUserStoreManager secondaryUserStoreManager = (AbstractUserStoreManager) carbonUM
                         .getSecondaryUserStoreManager(domainName);
-                totalUsers = secondaryUserStoreManager.countUsersWithClaims("http://wso2.org/claims/username", "*");
+                totalUsers = secondaryUserStoreManager.countUsersWithClaims(USERNAME_CLAIM, "*");
+            } catch (org.wso2.carbon.user.core.UserStoreException e) {
+                throw new CharonException("Error while getting total user count in domain: " + domainName);
+            }
+        } else if (StringUtils.isNotBlank(domainName)) {
```

Comment:
let's move this check to the beginning of the method and return 0 if domainname is null/empty. I.e if StringUtils.isBlank(domainName) return totalUsers;

Reply:
Then, if the domain name is null/empty and the PRIMARY userstore is a JDBC one, the totalResult is wrong